integrations/slack/webhooks/webhook.go

```diff
@@ -108,21 +111,56 @@ func checkRequest(w http.ResponseWriter, r *http.Request, l *zap.Logger, wantCon
 	}
 
 	// Request body.
-	b, err := io.ReadAll(r.Body)
+	body, err := io.ReadAll(r.Body)
 	if err != nil {
 		l.Error("Failed to read inbound HTTP request body",
 			zap.Error(err),
 		)
 		http.Error(w, "Bad Request", http.StatusBadRequest)
 		return nil
 	}
-	signingSecret := os.Getenv(signingSecretEnvVar)
-	if !verifySignature(signingSecret, ts, sig, b) {
-		l.Error("Slack signature verification failed")
+
+	appID, teamID, enterpriseID, err := h.extractIDs(body, wantContentType, l)
```

Comment:
Suggestion: split lines 123-148 into a helper function, e.g. "customOAuthSigningKey", to clarify the context of the entire block of code, and enable easier-to-read structure and comments without breaking the flow of reading the code.

Don't forget to add a header that explains the function's purpose in the larger context of `checkRequest`.